runelite-client/src/main/java/net/runelite/client/plugins/grandexchange/GrandExchangeSearchPanel.java

```diff
@@ -179,44 +258,39 @@ private void priceLookup(boolean exactMatch)
 				}
 			}
 		}
+		else
+		{
+			searchBox.setIcon(errorIcon);
+			errorPanel.setContent("No results found.", "No items were found with that name, please try again.");
+			resultsWrapper.setVisible(false);
+		}
 
 		SwingUtilities.invokeLater(() ->
 		{
-			if (ITEMS_LIST.isEmpty())
+
+			// unfortunately, as I have not found a automatic way to make the scroll pane resize
+			// to properly wrap the content, it has to recalculate its size every search
+			resultsWrapper.setPreferredSize(calculatePrefSize());
```

Comment:
The notes plugin may be doing something similar to what you are looking for. I think it sets a custom layout to the parent panel to get the scroll pane to update automatically.